assets/src/edit-story/components/library/panes/media/media3p/test/media3pPane.js

```diff
@@ -0,0 +1,181 @@
+};
+
+jest.mock('../../../../useLibrary');
+import useLibrary from '../../../../useLibrary';
+
+jest.mock('../../../../../../app/media/useMedia');
+import useMedia from '../../../../../../app/media/useMedia';
+
+jest.mock('../../../../../../app/media/media3p/providerConfiguration', () => ({
+  PROVIDERS: {
+    PROVIDER_1: {
+      displayName: 'Provider 1',
+      supportedContentTypes: ['image'],
+      supportsCategories: true,
+      requiresAuthorAttribution: true,
+      fetchMediaErrorMessage: 'Error loading media from Provider 1',
+      fetchCategoriesErrorMessage: 'Error loading categories from Provider 1',
+    },
+    PROVIDER_2: {
+      displayName: 'Provider 2',
+      supportedContentTypes: ['video'],
+      supportsCategories: false,
+      requiresAuthorAttribution: false,
+      fetchMediaErrorMessage: 'Error loading media from Provider 2',
+    },
+  },
+}));
+
+const createMediaResource = (name, provider) => ({
+  name,
+  provider: provider,
+  src: 'http://www.img.com/1',
+  width: 480,
+  height: 640,
+  imageUrls: [
+    {
+      imageName: 'full',
+      url: 'http://www.img.com/1',
+      width: 480,
+      height: 640,
+      mimeType: 'image/png',
+    },
+    {
+      imageName: 'large',
+      url: 'http://www.img.com/2',
+      width: 300,
+      height: 200,
+      mimeType: 'image/png',
+    },
+    {
+      imageName: 'web_stories_thumbnail',
+      url: 'http://www.img.com/3',
+      width: 200,
+      height: 100,
+      mimeType: 'image/png',
+    },
+  ],
+  description: 'A cat',
+  type: 'IMAGE',
+  createTime: '1234',
+  updateTime: '5678',
+});
+
+const DEFAULT_PROVIDER_STATE = (index) => ({
+  state: {
+    isMediaLoaded: false,
+    isMediaLoading: false,
+    hasMore: false,
+    media: [],
+    categories: {
+      selectedCategoryId: undefined,
+      categories: [
+        {
+          id: `provider${index}/1`,
+          displayName: `Tiny dogs for provider ${index}`,
+        },
+        {
+          id: `provider${index}/2`,
+          displayName: `Tiny cats for provider ${index}`,
+        },
+      ],
+    },
+  },
+  actions: {
+    selectCategory: jest.fn(),
+    deselectCategory: jest.fn(),
+    setNextPage: jest.fn(),
+  },
+});
+
+const DEFAULT_USE_MEDIA_RESULT = {
+  searchTerm: '',
+  selectedProvider: undefined,
+  setSelectedProvider: jest.fn(),
+  setSearchTerm: jest.fn(),
+  media3p: {
+    PROVIDER_1: DEFAULT_PROVIDER_STATE(1),
+    PROVIDER_2: DEFAULT_PROVIDER_STATE(2),
+  },
+};
+
+const MEDIA = Array(10).fill(createMediaResource('img', 'PROVIDER_1'));
+
+describe('Media3pPane', () => {
+  const insertElement = jest.fn();
+  let useMediaResult;
+  beforeAll(() => {
+    useLibrary.mockImplementation((selector) =>
+      selector({
+        actions: {
+          insertElement: insertElement,
+        },
+      })
+    );
+    useMedia.mockImplementation(() => useMediaResult);
+  });
+
+  beforeEach(() => {
+    useMediaResult = DEFAULT_USE_MEDIA_RESULT;
+  });
+
```

Comment:
should add a test for media-subheading to not be displayed when media hasn't loaded and no categories selected

Reply:
Done